src/middlewared/middlewared/api/v25_04_1/pool_dataset.py

```diff
@@ -0,0 +1,28 @@
+from typing import Union
+
+from middlewared.api.base import BaseModel
+
+
+__all__ = [
+    "PoolDatasetDestroySnapshotsArgs", "PoolDatasetDestroySnapshotsResult",
+]
+
+
+class PoolDatasetDestroySnapshotsArgs(BaseModel):
+    name: str
+    snapshots: "PoolDatasetDestroySnapshotsArgsSnapshots"
+
+
+class PoolDatasetDestroySnapshotsArgsSnapshots(BaseModel):
+    all: bool = False
+    recursive: bool = False
+    snapshots: list[Union["PoolDatasetDestroySnapshotsArgsSnapshotSpec", str]] = []
```

Comment:
Why not do it this way?
```suggestion
    snapshots: list[PoolDatasetDestroySnapshotsArgsSnapshotSpec, str]] = Field(default_factory=list)
```

Reply:
That'll be `NameError` because `PoolDatasetDestroySnapshotsArgsSnapshotSpec` is not defined yet.

`default_factory=list` is unnecessary https://docs.pydantic.dev/latest/concepts/fields/#mutable-default-values

Reply:
> That'll be `NameError` because `PoolDatasetDestroySnapshotsArgsSnapshotSpec` is not defined yet.
> 
Why not move the class definition after the other class's definition?

> `default_factory=list` is unnecessary https://docs.pydantic.dev/latest/concepts/fields/#mutable-default-values
>
Nice to know

Reply:
```suggestion
    snapshots: "list[PoolDatasetDestroySnapshotsArgsSnapshotSpec | str]" = []
```
is equivalent